src/query/procedure/module.cpp

```diff
@@ -320,6 +322,368 @@ void RegisterMgTransformations(const std::map<std::string, std::unique_ptr<Modul
+      return;
+    }
+
+    if (!std::filesystem::exists(path)) {
+      static_cast<void>(mgp_result_set_error_msg(result, "The specified file doesn't exist."));
+      return;
+    }
+
+    if (auto maybe_error_msg = ParentModuleDirectory(*module_registry, path); maybe_error_msg.HasError()) {
+      static_cast<void>(mgp_result_set_error_msg(result, maybe_error_msg.GetError()));
+      return;
+    }
+
+    auto *content_arg = Call<mgp_value *>(mgp_list_at, args, 1);
+    MG_ASSERT(CallBool(mgp_value_is_string, content_arg), "Should have been type checked already");
+    const char *content_str{nullptr};
+    {
+      const auto success = TryOrSetError([&] { return mgp_value_get_string(content_arg, &content_str); }, result);
+      if (!success) {
+        return;
+      }
+    }
+
+    if (auto maybe_error = WriteToFile(path, {content_str, std::strlen(content_str)}); maybe_error.HasError()) {
+      static_cast<void>(mgp_result_set_error_msg(result, maybe_error.GetError().c_str()));
+      return;
+    }
+
+    WithUpgradedLock(lock, [&]() { module_registry->UnloadAndLoadModulesFromDirectories(); });
+  };
+  mgp_proc update_module_file("update_module_file", std::move(update_module_file_cb), utils::NewDeleteResource(),
+                              false);
+  MG_ASSERT(mgp_proc_add_arg(&update_module_file, "path", Call<mgp_type *>(mgp_type_string)) == MGP_ERROR_NO_ERROR);
+  MG_ASSERT(mgp_proc_add_arg(&update_module_file, "content", Call<mgp_type *>(mgp_type_string)) == MGP_ERROR_NO_ERROR);
+  module->AddProcedure("update_module_file", std::move(update_module_file));
+}
+
+void RegisterMgDeleteModuleFile(ModuleRegistry *module_registry, utils::RWLock *lock, BuiltinModule *module) {
+  auto delete_module_file_cb = [module_registry, lock](mgp_list *args, mgp_graph * /*unused*/, mgp_result *result,
+                                                       mgp_memory * /*memory*/) {
+    MG_ASSERT(Call<size_t>(mgp_list_size, args) == 1U, "Should have been type checked already");
+    auto *path_arg = Call<mgp_value *>(mgp_list_at, args, 0);
+    MG_ASSERT(CallBool(mgp_value_is_string, path_arg), "Should have been type checked already");
+    const char *path_str{nullptr};
+    {
+      const auto success = TryOrSetError([&] { return mgp_value_get_string(path_arg, &path_str); }, result);
+      if (!success) {
+        return;
+      }
+    }
+
+    const std::filesystem::path path{path_str};
+
+    if (!path.is_absolute()) {
+      static_cast<void>(mgp_result_set_error_msg(result, "The path should be an absolute path."));
+      return;
+    }
+
+    if (!std::filesystem::exists(path)) {
+      static_cast<void>(mgp_result_set_error_msg(result, "The specified file doesn't exist."));
+      return;
+    }
+
+    if (!IsAllowedExtension(path.extension())) {
+      static_cast<void>(mgp_result_set_error_msg(result, "The specified file isn't in the supported format."));
+      return;
+    }
```

Comment:
Yesterday I somehow slipped through this, but the order should be also changed here in my opinion.